docs/PAYJOIN.md

```diff
@@ -213,144 +219,154 @@ bump the fee enough to add one input to the transaction, and this should be fine
-* The two sides communicate
+Joinmarket allows you to receive payments from any wallet that supports spending via the BIP78 Protocol, using a Tor hidden service.
 
-This takes generally only a couple of seconds, not including a relatively slow startup (30s-60s or so depending on configuration),
-during which time the sender acts similarly to any other Joinmarket participant (and so does the receiver, likewise), to have
-a slightly better "blend in with the crowd" effect.
+This hidden service is "ephemeral" meaning it is set up specifically for the payment and discarded after you shut down the receiving process. The setup takes some few seconds but it isn't too slow.
 
-Once the transaction is broadcast, you'll see a message to that effect and the script will shutdown. You can check the txid
-on your favourite block explorer.
+To make this work, you will need to do some minor configuring Tor, the first time. This is explained in detail [below](#torconfig). If you fail to do this step, you will see some rather unintelligible errors about connection failures when trying to run the script described next.
+
+Once that is ready, you can run the `receive-payjoin.py` script something like this:
+
+```python3
+(jmvenv)a$ python receive-payjoin.py -m1 walletname.jmdat 0.32
+```
 
-* What if something goes wrong and the payment fails to go through?
+The arguments provided are the wallet name and the exact amount you intend to receive (here it's 0.32 BTC (flagged as BTC because decimal), but you could also write `32000000` which will be interpreted as satoshis).
 
-First, if the sender succeeds in doing the first step, the receiver will have a non-coinjoin ordinary payment transaction
-to fall back on. It'll look like this:
+After a delay of 5-50 seconds (usually; Tor setup varies unpredictably), you will see a message like this:
 
 ```
-2019-01-16 16:34:55,168 [INFO]  Network transaction fee is: 1672 satoshis.
-2019-01-16 16:34:55,175 [INFO]  We'll use this serialized transaction to broadcast if your counterparty fails to broadcast the payjoin version:
-2019-01-16 16:34:55,177 [INFO]  0200000000010152da645b5a2ec3a166ad8a933b1442fa38fc119faf7659d72033ba863fdd8d470000000017160014297b55001daa905a3552137ef19755cf4eae7babfeffffff02b8be4f0a0000000017a914e3cb2bf72ccd4412035f9668e07e17b6c9ebd58d87c0fc9b010000000017a914b8bf57f3bae00d23f9a60c9a6e4d4c7182ec87cc8702483045022100e3605548f5e07ebd14a0700dc3e54e7c8ae90ce0057a3fdc80b5fd37636b44a002202fd7942104fb344b80b8fd91faf0394a0f274f7a68fa462084b8ecf2fa245a65012102f823d62891d8bc8544d4369bb98c6fb8235a372d2c36196d40c448690b42754f42030000
-2019-01-16 16:34:55,186 [INFO]  The proposed tx does not trigger UIH2, which means it is indistinguishable from a normal payment. This is the ideal case. Continuing..
-....
+Your hidden service is available. Please now pass this URI string to the sender to effect the payjoin payment:
+bitcoin:bc1xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx?amount=0.32000000&pj=http://fyrdc7dklhpify4g.onion:7083
```

Comment:
onion URL example here should be replaced with torv3 onion service URL, as torv2 onion services are being deprecated and in my tests it creates torv3 service anyway.

Reply:
Right. Unfortunately the tor that comes with my Ubuntu (1804) doesn't support the creation of v3 onions; I dug in and found the commit a few weeks back when doing this and it was after that release (I think you'll find details in the IRC log). I agree changing the example to v3 makes sense, will do.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/682/commits/68e013f5f83e91e3694924f0e9269f79360db082